src/components/action-popover/action-popover.component.tsx

```diff
@@ -23,34 +22,81 @@ import ActionPopoverItem from "./action-popover-item/action-popover-item.compone
 import ActionPopoverDivider from "./action-popover-divider/action-popover-divider.component";
 import ActionPopoverContext from "./action-popover-context";
 
-const marginPropTypes = filterStyledSystemMarginProps(
-  styledSystemPropTypes.space
-);
+interface RenderButtonProps {
+  tabIndex: number;
+  "data-element": string;
+  ariaAttributes: {
+    "aria-haspopup": string;
+    "aria-label": string;
+    "aria-controls": string;
+    "aria-expanded": string;
+  };
+}
+
+export interface ActionPopoverProps extends MarginProps {
+  /** Children for popover component */
+  children?: React.ReactNode;
+  /** Horizontal alignment of menu items content */
+  horizontalAlignment?: "left" | "right";
+  /** Unique ID */
+  id?: string;
+  /** Callback to be called on menu open */
+  onOpen?: () => void;
+  /** Callback to be called on menu close */
+  onClose?: () => void;
+  /** Set whether the menu should open above or below the button */
+  placement?: "bottom" | "top";
+  /** Render a custom menu button to override default ellipsis icon */
+  renderButton?: (buttonProps: RenderButtonProps) => React.ReactNode;
+  /** Boolean to control whether menu should align to right */
+  rightAlignMenu?: boolean;
+}
+
+const onOpenDefault = () => {};
+const onCloseDefault = () => {};
 
-const ActionPopover = ({
+export const ActionPopover = ({
   children,
   id,
-  onOpen,
-  onClose,
+  onOpen = onOpenDefault,
+  onClose = onCloseDefault,
   rightAlignMenu,
   renderButton,
   placement = "bottom",
-  horizontalAlignment,
+  horizontalAlignment = "left",
   ...rest
-}) => {
+}: ActionPopoverProps) => {
   const l = useLocale();
   const [isOpen, setOpenState] = useState(false);
   const [focusIndex, setFocusIndex] = useState(0);
   const [guid] = useState(createGuid());
-  const buttonRef = useRef();
-  const menu = useRef();
+  const buttonRef = useRef<HTMLDivElement>(null);
+  const menu = useRef<HTMLDivElement>(null);
 
   const itemCount = useMemo(() => {
-    return React.Children.toArray(children).filter(
-      (child) => child.type === ActionPopoverItem
-    ).length;
+    return React.Children.toArray(children).filter((child) => {
+      return React.isValidElement(child) && child.type === ActionPopoverItem;
+    }).length;
   }, [children]);
 
+  const hasIncorrectChildren = useMemo(() => {
+    return React.Children.toArray(children).find((child: React.ReactNode) => {
+      if (React.isValidElement(child)) {
+        return (
+          ActionPopoverItem !== child.type &&
+          ActionPopoverDivider !== child.type
+        );
+      }
+      return true;
+    });
+  }, [children]);
+
+  invariant(
+    !hasIncorrectChildren,
```

Comment:
double negation is a bit confusing, I think that better way to do it would be to flip the boolean in the method to be just `hasProperChildren`